plugins/package-managers/python/src/main/kotlin/Poetry.kt

```diff
@@ -72,6 +73,13 @@ class Poetry(
 
         return Pip(managerName, analysisRoot, analyzerConfig, repoConfig)
             .resolveDependencies(requirementsFile, labels)
+            .map { projectAnalyzerResult ->
+                projectAnalyzerResult.copy(
+                    project = projectAnalyzerResult.project.copy(
+                        definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path
```

Comment:
I propose `definitionFile.relativeTo(analysisRoot)` instead to avoid adding more code that relies on the project to be version controlled.

Reply:
This does not introduce the requirement, because code which computes the value for `definitionFilePath` already relies on that prior to this change. Furthermore, `relativeTo(analysisRoot)` seems wrong because the definition file path is not meant to be relative to the analysis root, but the repository root.

Reply:
> This does not introduce the requirement

And I didn't say that. I was talking about adding *more* code with that requirement.

> Furthermore, `relativeTo(analysisRoot)` seems wrong because the definition file path is not meant to be relative to the analysis root, but the repository root.

Indeed `Project.definitionFilePath` is documented like that 😞 IMO that's a mistake already, and it will make our lives harder in order to address https://github.com/oss-review-toolkit/ort/issues/2896.

Anyway, we already have a bunch of code like that, so we can bulk-address this later, I guess.

Reply:
> And I didn't say that. I was talking about adding _more_ code with that requirement.

I know, I didn't mean to say that you said that. I only intended to clarify that both, the old and the new code path depend on that same logic.